src/Native/Runtime/amd64/ExceptionHandling.S

```diff
@@ -0,0 +1,525 @@
+        // trash the values at the old homes to make sure nobody uses them
+        mov     rcx, 0xbaaddeed
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pRbx]
+        mov     [rax], rcx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pRbp]
+        mov     [rax], rcx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR12]
+        mov     [rax], rcx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR13]
+        mov     [rax], rcx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR14]
+        mov     [rax], rcx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR15]
+        mov     [rax], rcx
+#endif
+
+        mov     rdi, [rsi + OFFSETOF__REGDISPLAY__SP]                // rdi <- establisher frame
+        call    qword ptr [rsp + rsp_offsetof_arguments + 0x0]       // handler funclet address
+ALTERNATE_ENTRY RhpCallFinallyFunclet2
+
+        mov     rsi, [rsp + rsp_offsetof_arguments + 0x8]            // rsi <- regdisplay
+
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pRbx]
+        mov     [rax]                            , rbx
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pRbp]
+        mov     [rax]                            , rbp
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR12]
+        mov     [rax]                            , r12
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR13]
+        mov     [rax]                            , r13
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR14]
+        mov     [rax]                            , r14
+        mov     rax, [rsi + OFFSETOF__REGDISPLAY__pR15]
+        mov     [rax]                            , r15
+
+        mov     rax, [rsp + 0x20]                                    // rax <- Thread*
+        lock or             dword ptr [rax + OFFSETOF__Thread__m_ThreadStateFlags], ~TSF_DoNotTriggerGc
+
+        add     rsp, 0x28
+        pop     rbp
+        pop     rbx
+        pop     r12
+        pop     r13
+        pop     r14
+        pop     r15
+
+        ret
+
+NESTED_END RhpCallFinallyFunclet, _TEXT
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// void* FASTCALL RhpCallFilterFunclet(RtuObjectRef exceptionObj, void* pFilterIP, REGDISPLAY* pRegDisplay)
+//
+// INPUT:  RDI:  exception object
+//         RSI:  filter funclet address
+//         RDX:   REGDISPLAY*
+//
+// OUTPUT:
+// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+NESTED_ENTRY RhpCallFilterFunclet, _TEXT, NoHandler
+
+        push_nonvol_reg rbp     // we'll be setting rbp to the funclet's frame pointer
+
+        alloc_stack     0x20     // outgoing area
+
+        END_PROLOGUE
+
+        mov     rax, [rdx + OFFSETOF__REGDISPLAY__pRbp]
+        mov     rbp, [rax]
+
+        mov     rax, rsi                                            // rax <- handler funclet address
+#ifdef CORERT // @TODO Reconcile
```

Comment:
The ifdef is not necessary for the Unix version - just use the CORERT path. (There is plan underway to switch Windows to CORERT path unconditionally as well.)